### PR TITLE
N+1 queries

### PR DESCRIPTION
**Pull request by Sahar-SE**
### This PR Contains:

- Solve N+1 queries at Recipe
- Solve N+1 queries at Public Recipes
- Solve N+1 queries at Food Recipes
- Update Navbar
- Update Registration

### CODE REVIEW

File: app/models/food.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class Food < ApplicationRecord
-  belongs_to :user
-  # has_many :receipFoods
+  belongs_to :users
```
Comment (shadabwahidullah) [inline app/models/food.rb]: Please change the users to user a food only belongs to a single user

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -5,16 +5,19 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-    <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
```
Comment (shadabwahidullah) [inline app/views/layouts/application.html.erb:8]: This line should not be removed. it's related to tailwind configuration

File: config/routes.rb [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
   # Defines the root path route ("/")
   # root "articles#index"
-  resources :users
+
+  root "recipes#public"
+  resources :users do
+    resources :recipes
+  end
   resources :foods
   resources :recipes
```
Comment (shadabwahidullah) [inline config/routes.rb:14]: Here I believe we only need the routes to recipes to be defined once. Why do you think recipes routes should be nested inside users?

**Review by shadabwahidullah — CHANGES_REQUESTED** [review]
# Hi @Sahar-SE , 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍 
- CRUD feature is functioning as intended
- Functions effectively split into various files
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

#### Optional suggestions
 
 - N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: config/routes.rb (hunk 3 above)
Comment (Sahar-SE) [reply config/routes.rb:14]: @shadabwahidullah Thanks😊

**Review by shadabwahidullah — APPROVED** [review]
## Hi @Sahar-SE 

# Status: Approved ✔️

<strong> Congratulations! 🎉  </strong>   You have implemented all the requested changes. Well Done.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Wahidullah Shadab_**

**Pull request merged**